conftest.py

```diff
@@ -3,8 +3,54 @@
 from enum import Enum
 import requests
 from typing import Callable
+import logging
+from dataclasses import dataclass
 
+log = logging.getLogger(__name__)
 
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int
+
+    def __post_init__(self):
+        if not isinstance(self.name, str):
+            raise TypeError("'name' should be a string")
+        
+        if self.name == "":
+            raise ValueError("'name' should not be empty")
+        
+
+        if not isinstance(self.hid, str):
+            raise TypeError("'hid' should be a string")
+
+        if self.hid == "":
+            raise ValueError("'hid' should not be empty")
+        
+
+        if not isinstance(self.model, str):
+            raise TypeError("'model' should be a string")
+
+        if self.model == "":
+            raise ValueError("'model' should not be empty")
+
+
+        if not isinstance(self.firmware_version, int):
+            raise TypeError("'firmware_version' should be an integer")
+
+        if self.firmware_version == "":
```

Comment:
Testing for a non-empty string is suitable for string properties but here we've got an integer. We'd like to test this for validity from sensor's standpoint. That would be checking that the firmware version is between 10 and 15, both ends included. You could test this with the following code:
```
10 <= self.firmware_version <= 15
```
Of course, you'll need to update the error message in the `ValueError` to reflect this new check.